tests/storage/blocksd_test.py

```diff
@@ -1244,6 +1251,117 @@ def test_create_volume_with_new_bitmap(
     ]
 
 
+@requires_root
+@pytest.mark.root
+@pytest.mark.xfail(reason='Bitmaps in base volume not considered for merge '
+                          'if added after top volume is created.')
+def test_merge_with_bitmap(
+        tmp_storage, tmp_repo, fake_access, fake_rescan, tmp_db, fake_task,
+        fake_sanlock, monkeypatch):
+    sd_id = str(uuid.uuid4())
+
+    dev = tmp_storage.create_device(10 * GiB)
+    lvm.createVG(sd_id, [dev], blockSD.STORAGE_UNREADY_DOMAIN_TAG, 128)
+    vg = lvm.getVG(sd_id)
+
+    dom = blockSD.BlockStorageDomain.create(
+        sdUUID=sd_id,
+        domainName="domain",
+        domClass=sd.DATA_DOMAIN,
+        vgUUID=vg.uuid,
+        version=4,
+        storageType=sd.ISCSI_DOMAIN)
+
+    sdCache.knownSDs[sd_id] = blockSD.findDomain
+    sdCache.manuallyAddDomain(dom)
+
+    dom.refresh()
+    dom.attach(tmp_repo.pool_id)
+
+    img_id = str(uuid.uuid4())
+    base_vol_uuid = str(uuid.uuid4())
+    top_vol_uuid = str(uuid.uuid4())
+
+    # Create base volume.
+    dom.createVolume(
+        imgUUID=img_id,
+        capacity=1 * GiB,
+        volFormat=sc.COW_FORMAT,
+        preallocate=sc.SPARSE_VOL,
+        diskType='DATA',
+        volUUID=base_vol_uuid,
+        desc="Test volume",
+        srcImgUUID=sc.BLANK_UUID,
+        srcVolUUID=sc.BLANK_UUID,
+        bitmap="new-bitmap")
+
+    base_vol = dom.produceVolume(img_id, base_vol_uuid)
+
+    # Create top volume with bitmaps.
+    dom.createVolume(
+        imgUUID=img_id,
+        capacity=1 * GiB,
+        volFormat=sc.COW_FORMAT,
+        preallocate=sc.SPARSE_VOL,
+        diskType='DATA',
+        volUUID=top_vol_uuid,
+        desc="Test top volume",
+        srcImgUUID=base_vol.imgUUID,
+        srcVolUUID=base_vol.volUUID,
+        add_bitmaps=True)
```

Comment:
Why use add_bitmaps? this creates invalid situation since the leaf has no bitmaps.
In a normal chain, bitmaps are always in the child volumes. For example, bitmap
can be missing from base but it must be in top and leaf.